Orleans/src/Microsoft.Orleans.Host.Abstractions/AzureStorageProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    // this would go into the azure storage providers DLL
+    public static class AzureStorageProviderExtensions
+    {
+        public static void AddAzureStorage(this IStorageProvidersBuilder factory, string providerName, string connectionString)
```

Comment:
Why isn't this method member of `AzureStorageProviderFactory`? 

If you worry about `AzureStorageProviderFactory's `accessability  as an internal class, maybe we can use an interface as a medium to publicize `AddAzureStorage` method:  
`interface IAzureStorageProviderFactory:
{
    public AddAzureStorage(...)
}`